ocs_ci/ocs/longevity.py

```diff
@@ -554,6 +809,313 @@ def stage_0(
+                                access_modes=access_modes,
+                                status=constants.STATUS_BOUND,
+                                num_of_pvc=num_of_pvc,
+                                wait_each=True,
+                            )
+                            log.info(
+                                f"PVCs of interface:{interface} for operation:{operation} were successfully created."
+                            )
+                            pvc_objs.extend(pvc_objs_tmp)
+                        else:
+                            log.error(
+                                f"Num of PVCs of interface - {interface} = {num_of_pvc}. So no PVCs created."
+                            )
+
+                    # Create PODs
+                    pod_objs = list()
+                    for pvc_obj in pvc_objs:
+                        if pvc_obj.get_pvc_vol_mode == constants.VOLUME_MODE_BLOCK:
+                            pod_objs.append(
+                                pod_factory(
+                                    pvc=pvc_obj,
+                                    raw_block_pv=True,
+                                    status=constants.STATUS_RUNNING,
+                                    pod_dict_path=constants.PERF_BLOCK_POD_YAML,
+                                )
+                            )
+                        else:
+                            pod_objs.append(
+                                pod_factory(
+                                    pvc=pvc_obj,
+                                    status=constants.STATUS_RUNNING,
+                                    pod_dict_path=constants.PERF_POD_YAML,
+                                )
+                            )
+
+                    log.info(
+                        f"PODs for operation:{operation} were successfully created."
+                    )
+
+                    # Run IO to utilize 25% of volume
+                    log.info("Run IO on all pods to utilise 25% of PVCs")
+                    file_name = "fio_25"
+                    for pod_obj in pod_objs:
+                        log.info(f"Running IO on pod {pod_obj.name}")
+                        log.info(f"File created during IO {file_name}")
+                        fio_size = int(0.25 * pvc_size * 1000)
+                        storage_type = (
+                            "block"
+                            if pod_obj.pvc.get_pvc_vol_mode
+                            == constants.VOLUME_MODE_BLOCK
+                            else "fs"
+                        )
+                        pod_obj.wl_setup_done = True
+                        pod_obj.wl_obj = _workload.WorkLoad(
+                            "test_workload_fio",
+                            pod_obj.get_storage_path(storage_type),
+                            "fio",
+                            storage_type,
+                            pod_obj,
+                            1,
+                        )
```

Comment:
why do we need these lines of code? direct call of pod_obj.run_io doesnt work?

Reply:
Without these lines also, pod_obj.run_io would work; just that it will try to install fio again. To avoid that we have added these lines, in the same way as we did in stage2.